README.md

```diff
@@ -97,6 +97,10 @@ change the way Spack concretizes dependencies within a release branch.
 So, you can base your Spack deployment on a release branch and `git pull`
 to get fixes, without the package churn that comes with `develop`.
 
+The latest release is always available on the `releases/latest` branch.
```

Comment:
Wondering if we should use:
```suggestion
The highest release is always available on the `releases/latest` branch.
```
(as we do in the documentation) to avoid misunderstanding in case we cut a point release of a previous major version e.g. if we release v0.14.3 after having released v0.15.0. In any case docs afterwards have all the details.